src/SearchParameters/RefinementList.js

```diff
@@ -94,25 +91,27 @@ var lib = {
    * @return {RefinementList} a new and updated refinement list
    */
   clearRefinement: function clearRefinement(refinementList, attribute, refinementType) {
-    if (isUndefined(attribute)) {
-      if (isEmpty(refinementList)) return refinementList;
+    if (attribute === undefined) {
+      if (Object.keys(refinementList).length === 0) return refinementList;
       return {};
     } else if (typeof attribute === 'string') {
-      if (isEmpty(refinementList[attribute])) return refinementList;
+      if (refinementList[attribute] === undefined) return refinementList;
```

Comment:
This is not similar. Is that expected?

Reply:
isEmpty will also be true for undefined

Reply:
according to the tests at least, it's either present in the object, either undefined.